server/src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java

```diff
@@ -599,12 +609,132 @@ public void testDevPoolRemovalAtUnbind() throws EntitlementRefusedException {
         AutobindData ad = new AutobindData(devSystem);
         ad.setPossiblePools(possPools);
         List<Entitlement> results = poolManager.entitleByProducts(ad);
+        // TODO fix me.This should not be necessary
+        poolCurator.clear();
         assertEquals(1, results.size());
         assertEquals(results.get(0).getPool(), pool1);
 
-        poolManager.revokeEntitlement(results.get(0));
+        Entitlement e = entitlementCurator.find(results.get(0).getId());
+        poolManager.revokeEntitlement(e);
         assertNull(poolCurator.find(pool1.getId()));
         assertNotNull(poolCurator.find(pool2.getId()));
     }
+
+    @Test
+    public void testDevPoolBatchBind() throws EntitlementRefusedException {
+        Owner owner = createOwner();
+        Product p = new Product("test-product", "Test Product", owner);
+        productCurator.create(p);
+
+        Consumer devSystem = new Consumer("dev", "user", owner, systemType);
+        devSystem.setFact("dev_sku", p.getId());
+        devSystem.addInstalledProduct(new ConsumerInstalledProduct(p));
+        consumerCurator.create(devSystem);
+
+        Pool pool1 = createPool(owner, p, 10L, TestUtil.createDate(2000, 3, 2),
+                TestUtil.createDate(2050, 3, 2));
+        pool1.addAttribute(new PoolAttribute(Pool.DEVELOPMENT_POOL_ATTRIBUTE, "true"));
+        pool1.addAttribute(new PoolAttribute(Pool.REQUIRES_CONSUMER_ATTRIBUTE, devSystem.getUuid()));
+        poolCurator.create(pool1);
+        Pool pool2 = createPool(owner, p, 10L, TestUtil.createDate(2000, 3, 2),
+                TestUtil.createDate(2050, 3, 2));
+        poolCurator.create(pool2);
+
+        Map<String, Integer> poolQuantities = new HashMap<String, Integer>();
+        poolQuantities.put(pool1.getId(), 1);
+        poolQuantities.put(pool2.getId(), 1);
+
+        List<Entitlement> results = poolManager.entitleByPools(devSystem, poolQuantities);
+        // TODO fix me.This should not be necessary
```

Comment:
The same as above, hopefuly after the implementation is fixed, this will not be necessary